File: Ejercicios/JavaScript/Clases/Producto.js
class Producto {
  constructor(nombre, precio) {
    if (nombre === undefined && precio === undefined) {
      this.nombre = "Desconocido";
      this.precio = 0;
    } else if (nombre instanceof Producto) {
      this.nombre = nombre.nombre;
      this.precio = nombre.precio;
    } else if (precio === undefined) {
      this.nombre = nombre;
      this.precio = 10;
    } else {
      this.nombre = nombre;
      this.precio = precio;
    }
  }
}

export default Producto;

File: Ejercicios/GO/Clases/Producto.go
package main

type Producto struct {
	Nombre string
	Precio float64
}

func NewProducto() Producto {
	return Producto{Nombre: "Desconocido", Precio: 0}
}

func NewProductoConDatos(nombre string, precio float64) Producto {
	return Producto{Nombre: nombre, Precio: precio}
}

func NewProductoNombre(nombre string) Producto {
	return Producto{Nombre: nombre, Precio: 10}
}

func NewProductoCopy(p Producto) Producto {
	return Producto{Nombre: p.Nombre, Precio: p.Precio}
}
